<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>888</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="verticalLayoutWidget_7">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>30</y>
      <width>501</width>
      <height>411</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_7">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>L1</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QSpinBox" name="L1_len"/>
           </item>
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>len</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QSpinBox" name="L1_block"/>
           </item>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>block</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <item>
            <widget class="QSpinBox" name="L1_time"/>
           </item>
           <item>
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>time</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLCDNumber" name="L1_size">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>L2</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QSpinBox" name="L2_len"/>
           </item>
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>len</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QSpinBox" name="L2_block"/>
           </item>
           <item>
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>block</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_9">
           <item>
            <widget class="QSpinBox" name="L2_time"/>
           </item>
           <item>
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>time</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLCDNumber" name="L2_size">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>L3</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QSpinBox" name="L3_len"/>
           </item>
           <item>
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>len</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QSpinBox" name="L3_block"/>
           </item>
           <item>
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>block</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_10">
           <item>
            <widget class="QSpinBox" name="L3_time"/>
           </item>
           <item>
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>time</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLCDNumber" name="L3_size">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_14">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string> translate table</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_13">
           <item>
            <widget class="QSpinBox" name="t_table_len"/>
           </item>
           <item>
            <widget class="QLabel" name="label_18">
             <property name="text">
              <string>len</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_19">
           <item>
            <widget class="QSpinBox" name="t_table_time"/>
           </item>
           <item>
            <widget class="QLabel" name="label_23">
             <property name="text">
              <string>time</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLCDNumber" name="t_table_size"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>TLB</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_12">
           <item>
            <widget class="QSpinBox" name="tlb_len"/>
           </item>
           <item>
            <widget class="QLabel" name="label_16">
             <property name="text">
              <string>len</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_20">
           <item>
            <widget class="QSpinBox" name="tlb_time"/>
           </item>
           <item>
            <widget class="QLabel" name="label_24">
             <property name="text">
              <string>time</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLCDNumber" name="tlb_size"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QLabel" name="label_13">
           <property name="text">
            <string>main memory</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_11">
           <item>
            <widget class="QSpinBox" name="mem_len"/>
           </item>
           <item>
            <widget class="QLabel" name="label_14">
             <property name="text">
              <string>len</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_21">
           <item>
            <widget class="QSpinBox" name="main_mem_time"/>
           </item>
           <item>
            <widget class="QLabel" name="label_25">
             <property name="text">
              <string>time</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLCDNumber" name="mem_size"/>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_15">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_9">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_17">
           <item>
            <widget class="QSlider" name="p_locality">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="invertedAppearance">
              <bool>false</bool>
             </property>
             <property name="invertedControls">
              <bool>false</bool>
             </property>
             <property name="tickPosition">
              <enum>QSlider::TickPosition::TicksAbove</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_21">
             <property name="text">
              <string>partial locality</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_18">
           <item>
            <widget class="QSlider" name="t_locality">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="invertedAppearance">
              <bool>false</bool>
             </property>
             <property name="tickPosition">
              <enum>QSlider::TickPosition::TicksAbove</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_22">
             <property name="text">
              <string>temporal locality</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_16">
           <item>
            <widget class="QSpinBox" name="address_count"/>
           </item>
           <item>
            <widget class="QLabel" name="label_20">
             <property name="text">
              <string>address count</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="generattor_btn">
           <property name="text">
            <string>generate new addresses</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_8">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_22">
           <item>
            <widget class="QSpinBox" name="disk_time"/>
           </item>
           <item>
            <widget class="QLabel" name="label_26">
             <property name="text">
              <string>disk time</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="label_19">
           <property name="text">
            <string> choose cache policy:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="cpt">
           <item>
            <property name="text">
             <string>LRU</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>MRU</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>LFU</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>FIFO</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>RR</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="run_btn">
           <property name="text">
            <string>Run !!!</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>888</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
